app/src/main/java/com/github/houseorganizer/houseorganizer/EditHousehold.java

```diff
@@ -2,60 +2,203 @@
+                    Map<String, Object> householdData = task.getResult().getData();
+                    if(householdData != null) {
+                        List<String> listOfUsers =
+                                (List<String>) householdData.getOrDefault("residents", "[]");
+                        if(listOfUsers.contains(email)){
+                            currentHousehold.update("owner", email);
+
+                            Toast.makeText(getApplicationContext(),
+                                    view.getContext().getString(R.string.owner_change_success),
+                                    Toast.LENGTH_SHORT).show();
+                            confirmChanges(view); //User is not owner anymore
+                        }
+                    }
+                });
+    }
+
+    public void removeUser(View view) {
+        TextView emailView = findViewById(R.id.editTextRemoveUser);
+        if(!verifyEmailInput(emailView, view)){
+            return;
+        }
+        String email = emailView.getText().toString();
+
+        if(email.equals(mAuth.getCurrentUser().getEmail().toString())){
+            Toast.makeText(getApplicationContext(),
+                    view.getContext().getString(R.string.cant_remove_yourself),
+                    Toast.LENGTH_SHORT).show();
+            return;
+        }
+
+        Task<SignInMethodQueryResult> signInMethodQueryResultTask =
+                mAuth.fetchSignInMethodsForEmail(email);
+
+        signInMethodQueryResultTask
+                .addOnCompleteListener(task -> {
+                    List<String> signInMethods = task.getResult().getSignInMethods();
+                    if(signInMethods != null && signInMethods.size() > 0){
+                        removeUserFromHousehold(email, view);
+                    }
+                });
+    }
+
+    public void removeUserFromHousehold(String email, View view){
+        firestore.collection("households").document(householdId).get()
+                 .addOnCompleteListener(task -> {
+                     Map<String, Object> householdData = task.getResult().getData();
+                     if(householdData != null) {
+                         List<String> listOfUsers =
+                                 (List<String>) householdData.getOrDefault("residents", "[]");
+                         Long num_users = (Long) householdData.get("num_members");
+                         if(listOfUsers.contains(email)){
+                             currentHousehold.update("residents", FieldValue.arrayRemove(email));
+                             currentHousehold.update("num_members",num_users-1);
+
+                             Toast.makeText(getApplicationContext(),
+                                     view.getContext().getString(R.string.remove_user_success),
+                                     Toast.LENGTH_SHORT).show();
+                         }
+                     }
+                 });
+    }
+
```

Comment:
The methods names are a little confusing but im really not sure what to rename them to

Reply:
I think the same but code climate made me create lots of (I think) useless methods otherwise it says that I duplicate code

Reply:
Perhaps we can find better names as part of a more general refactoring, but that's a task for future sprints.